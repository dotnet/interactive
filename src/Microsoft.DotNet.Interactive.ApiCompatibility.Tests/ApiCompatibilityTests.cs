// Copyright (c) .NET Foundation and contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System.Diagnostics;
using Assent;
using Microsoft.DotNet.Interactive.CSharp;
using Microsoft.DotNet.Interactive.Documents;
using Microsoft.DotNet.Interactive.Formatting;
using Microsoft.DotNet.Interactive.FSharp;
using Microsoft.DotNet.Interactive.Journey;
using Microsoft.DotNet.Interactive.Kql;
using Microsoft.DotNet.Interactive.PowerShell;
using Microsoft.DotNet.Interactive.SqlServer;
using Microsoft.DotNet.Interactive.Tests.Utility;
using Xunit;

namespace Microsoft.DotNet.Interactive.ApiCompatibility.Tests;

public class ApiCompatibilityTests
{
    private readonly Configuration _configuration;

    public ApiCompatibilityTests()
    {
        _configuration = new Configuration()
                         .SetInteractive(Debugger.IsAttached)
                         .UsingExtension("txt");
    }

    [FactSkipLinux]
    public void Interactive_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<Kernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void Formatting_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<FormatContext>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void Document_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<InteractiveDocument>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void PackageManagement_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<PackageRestoreContext>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void Journey_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<Lesson>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void csharp_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<CSharpKernel>();
        this.Assent(contract, _configuration);
    }

    [Fact(Skip = "need to use signature files")]
    public void fsharp_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<FSharpKernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void powershell_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<PowerShellKernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void mssql_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<MsSqlKernelConnector>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void kql_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<KqlKernelConnector>();
        this.Assent(contract, _configuration);
    }
}