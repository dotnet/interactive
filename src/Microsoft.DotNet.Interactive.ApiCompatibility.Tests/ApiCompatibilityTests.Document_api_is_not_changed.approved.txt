Microsoft.DotNet.Interactive.Documents
  public static class CodeSubmission
    public static System.Text.Encoding Encoding { get;}
    public static InteractiveDocument Parse(System.String content, KernelInfoCollection kernelInfo = null)
    public static InteractiveDocument Read(System.IO.Stream stream, KernelInfoCollection kernelInfos)
    public static System.Threading.Tasks.Task<InteractiveDocument> ReadAsync(System.IO.Stream stream, KernelInfoCollection kernelInfos)
    public static System.String ToCodeSubmissionContent(System.String newline = 
)
    public static System.Void Write(InteractiveDocument document, System.IO.Stream stream, System.String newline = 
)
    public static System.Void Write(InteractiveDocument document, System.IO.TextWriter writer, System.String newline = 
)
  public class DisplayElement : InteractiveDocumentOutputElement, IDataElement
    .ctor(System.Collections.Generic.IDictionary<System.String,System.Object> data)
    public System.Collections.Generic.IDictionary<System.String,System.Object> Data { get;}
    public System.Collections.Generic.IDictionary<System.String,System.Object> Metadata { get;}
  public class ErrorElement : InteractiveDocumentOutputElement
    .ctor(System.String errorValue = null, System.String errorName = Error, System.String[] stackTrace = null)
    public System.String ErrorName { get;}
    public System.String ErrorValue { get;}
    public System.String[] StackTrace { get;}
  public class InputField
    .ctor(System.String valueName, System.String typeHint = text)
    public System.String TypeHint { get; set;}
    public System.String ValueName { get; set;}
    protected System.Boolean Equals(InputField other)
    public System.Boolean Equals(System.Object obj)
    public System.Int32 GetHashCode()
  public class InteractiveDocument, System.Collections.IEnumerable
    public static System.Threading.Tasks.Task<InteractiveDocument> LoadAsync(System.IO.FileInfo file, KernelInfoCollection kernelInfos)
    .ctor(System.Collections.Generic.IList<InteractiveDocumentElement> elements = null)
    public System.Collections.Generic.IList<InteractiveDocumentElement> Elements { get;}
    public System.Collections.Generic.IDictionary<System.String,System.Object> Metadata { get;}
    public System.Void Add(InteractiveDocumentElement element)
    public System.String GetDefaultKernelName()
    public System.Collections.IEnumerator GetEnumerator()
    public System.Collections.Generic.IAsyncEnumerable<InteractiveDocument> GetImportsAsync(System.Boolean recursive = False)
    public System.Collections.Generic.IEnumerable<InputField> GetInputFields()
    public System.Collections.Generic.IEnumerable<System.String> GetMagicCommandLines()
  public class InteractiveDocumentElement
    .ctor()
    .ctor(System.String contents = null, System.String kernelName = null, System.Collections.Generic.IEnumerable<InteractiveDocumentOutputElement> outputs = null)
    public System.String Contents { get; set;}
    public System.Int32 ExecutionOrder { get; set;}
    public System.String Id { get; set;}
    public System.String KernelName { get; set;}
    public System.Collections.Generic.IDictionary<System.String,System.Object> Metadata { get; set;}
    public System.Collections.Generic.List<InteractiveDocumentOutputElement> Outputs { get;}
  public abstract class InteractiveDocumentOutputElement
  public class KernelInfo
    .ctor(System.String name, System.Collections.Generic.IReadOnlyCollection<System.String> aliases = null)
    public System.Collections.Generic.IReadOnlyCollection<System.String> Aliases { get;}
    public System.String Name { get;}
    public System.String ToString()
  public class KernelInfoCollection, System.Collections.Generic.ICollection<KernelInfo>, System.Collections.Generic.IEnumerable<KernelInfo>, System.Collections.IEnumerable
    .ctor()
    public System.Int32 Count { get;}
    public System.String DefaultKernelName { get; set;}
    public System.Boolean IsReadOnly { get;}
    public System.Void Add(KernelInfo kernelInfo)
    public System.Void AddRange(System.Collections.Generic.IEnumerable<KernelInfo> collection)
    public System.Void Clear()
    public KernelInfoCollection Clone()
    public System.Boolean Contains(KernelInfo kernelInfo)
    public System.Boolean Contains(System.String nameOrAlias)
    public System.Void CopyTo(KernelInfo[] array, System.Int32 arrayIndex)
    public System.Boolean Remove(KernelInfo item)
    public System.Boolean TryGetByAlias(System.String alias, ref KernelInfo& info)
  public class ReturnValueElement : InteractiveDocumentOutputElement, IDataElement
    .ctor(System.Collections.Generic.IDictionary<System.String,System.Object> data)
    public System.Collections.Generic.IDictionary<System.String,System.Object> Data { get;}
    public System.Int32 ExecutionOrder { get; set;}
    public System.Collections.Generic.IDictionary<System.String,System.Object> Metadata { get;}
  public class TextElement : InteractiveDocumentOutputElement
    .ctor(System.String text, System.String name)
    public System.String Name { get;}
    public System.String Text { get;}
Microsoft.DotNet.Interactive.Documents.Jupyter
  public class InputCellMetadata
    .ctor(System.String language = null)
    public System.String Language { get;}
  public static class InteractiveDocumentExtensions
    public static Microsoft.DotNet.Interactive.Documents.InteractiveDocument WithJupyterMetadata(System.String language = C#)
  public static class Notebook
    public static System.Text.Encoding Encoding { get;}
    public static System.Text.Json.JsonSerializerOptions JsonSerializerOptions { get;}
    public static Microsoft.DotNet.Interactive.Documents.InteractiveDocument Parse(System.String json, Microsoft.DotNet.Interactive.Documents.KernelInfoCollection kernelInfo = null)
    public static Microsoft.DotNet.Interactive.Documents.InteractiveDocument Read(System.IO.Stream stream, Microsoft.DotNet.Interactive.Documents.KernelInfoCollection kernelInfos)
    public static System.Threading.Tasks.Task<Microsoft.DotNet.Interactive.Documents.InteractiveDocument> ReadAsync(System.IO.Stream stream, Microsoft.DotNet.Interactive.Documents.KernelInfoCollection kernelInfo = null)
    public static System.String ToJupyterJson(System.String defaultLanguage = null)
    public static System.Void Write(Microsoft.DotNet.Interactive.Documents.InteractiveDocument document, System.IO.Stream stream)
    public static System.Void Write(Microsoft.DotNet.Interactive.Documents.InteractiveDocument document, System.IO.TextWriter writer)
Microsoft.DotNet.Interactive.Documents.ParserServer
  public enum DocumentSerializationType : System.Enum, System.IComparable, System.IConvertible, System.IFormattable
    Dib=0
    Ipynb=1
  public class NotebookErrorResponse : NotebookParserServerResponse
    .ctor(System.String id, System.String errorMessage)
    public System.String ErrorMessage { get;}
  public abstract class NotebookParseOrSerializeRequest
    public static NotebookParseOrSerializeRequest FromJson(System.String json)
    public System.String DefaultLanguage { get;}
    public System.String Id { get;}
    public DocumentSerializationType SerializationType { get;}
    public RequestType Type { get;}
  public class NotebookParseRequest : NotebookParseOrSerializeRequest
    .ctor(System.String id, DocumentSerializationType serializationType, System.String defaultLanguage, System.Byte[] rawData)
    public System.Byte[] RawData { get;}
    public RequestType Type { get;}
  public class NotebookParseResponse : NotebookParserServerResponse
    .ctor(System.String id, Microsoft.DotNet.Interactive.Documents.InteractiveDocument document)
    public Microsoft.DotNet.Interactive.Documents.InteractiveDocument Document { get;}
  public class NotebookParserServer, System.IDisposable
    public static NotebookParserServerResponse HandleRequest(NotebookParseOrSerializeRequest request)
    .ctor(System.IO.TextReader input, System.IO.TextWriter output)
    public System.IO.TextReader Input { get;}
    public System.IO.TextWriter Output { get;}
    public System.Void Dispose()
    public System.Threading.Tasks.Task RunAsync()
  public static class NotebookParserServerExtensions
    public static System.String ToJson()
    public static System.String ToJson()
  public abstract class NotebookParserServerResponse
    public static NotebookParserServerResponse FromJson(System.String json)
    public System.String Id { get;}
  public class NotebookSerializeRequest : NotebookParseOrSerializeRequest
    .ctor(System.String id, DocumentSerializationType serializationType, System.String defaultLanguage, System.String newLine, Microsoft.DotNet.Interactive.Documents.InteractiveDocument document)
    public Microsoft.DotNet.Interactive.Documents.InteractiveDocument Document { get;}
    public System.String NewLine { get;}
    public RequestType Type { get;}
  public class NotebookSerializeResponse : NotebookParserServerResponse
    .ctor(System.String id, System.Byte[] rawData)
    public System.Byte[] RawData { get;}
  public static class ParserServerSerializer
    public static System.Text.Json.JsonSerializerOptions JsonSerializerOptions { get;}
  public enum RequestType : System.Enum, System.IComparable, System.IConvertible, System.IFormattable
    Parse=0
    Serialize=1
